src/parser/GraphPatternOperation.h

```diff
@@ -43,6 +43,25 @@ struct SparqlValues {
   std::string valuesToString() const;
 };
 
+/// A `SERVICE` clause.
+struct Service {
+ public:
+  // The visible variables of the service clause.
+  std::vector<Variable> visibleVariables_;
+  // The URL of the service clause.
+  Iri serviceIri_;
+  // The prologue (prefix definitions).
+  std::string prologue_;
+  // The body of the SPARQL query for the remote endpoint.
+  std::string graphPatternAsString_;
+  // The corresponding graph pattern from the parser.
+  //
+  // TODO: This is currently only used by `GraphPatternOperation::toString()`.
+  // What is the use of that function? Does it really need the `GraphPattern`
+  // object from the partse or would `graphPatternAsString_` do equally fine?
```

Comment:
As far as I understand it, that function is only used to create readable
output for failed unit tests, so it's perfectly fine to only store the string here.